Gravatar/GravatarService.cs

```diff
@@ -137,18 +149,54 @@ private static string HashEmail(string email)
             return MD5.CalcMD5(email.Trim().ToLowerInvariant());
         }
 
-        private async Task<Image> LoadFromGravatarAsync(string imageFileName, string email, int imageSize, DefaultImageType defaultImageType)
+        private async Task<Image> LoadFromGitHubAsync(string imageFileName, string email, int imageSize)
+        {
+            string imageUrl = null;
+            try
+            {
+                int suffixPosition = email.IndexOf(GitHubPrivateEmailSuffix, StringComparison.OrdinalIgnoreCase);
+                string username = email.Substring(0, suffixPosition);
+                var client = new Git.hub.Client();
+                var user = client.GetUser(username);
```

Comment:
How is failed GH login supposed to be handled? The handling seem to indicate that no avatar at all is handled, which removes functionality for users. (Could be acceptable.)

Reply:
There's no login, this API is accessed anonymously. Do you mean if the GH username is not found? In this case it will return null, so there will be no avatar.

Reply:
For me, the Git.hub.dll is never loaded, I have not found why. This should rase an exception.

Reply:
> For me, the Git.hub.dll is never loaded, I have not found why

It works fine for me... Git.hub.dll is used in several other places, so it should work. Not sure what could be causing this.

> This should rase an exception.

What should raise an exception?

Reply:
The dll is not loaded, so the call will fail

Reply:
I'm not sure I understand what you mean. It's not a "normal" exception. The DLL is referenced, so it should be loaded the first time it's used. If it's not loaded, it's because of a deployment issue, not a code issue.

Anyway, if this happens, the error will be caught, and the method will return null.

Reply:
I agree, a referenced dll should load.
Failure to load is most likely a symptom, not a cause